src/ln/msgs.rs

```diff
@@ -585,6 +586,9 @@ pub trait RoutingMessageHandler : Send + Sync {
 	fn handle_channel_update(&self, msg: &ChannelUpdate) -> Result<bool, HandleError>;
 	/// Handle some updates to the route graph that we learned due to an outbound failed payment.
 	fn handle_htlc_fail_channel_update(&self, update: &HTLCFailChannelUpdate);
+	///Gets a subset of the messages required to dump our routing table to a remote node, starting at the short_channel_id indicated by starting_point and including batch_amount entries
+	/// This function will start iterating at 0 if the starting_point is < 0.
+	fn get_next_announcements(&self, starting_point: i64, batch_amount: u8)->(Vec<(ChannelAnnouncement, ChannelUpdate, NodeAnnouncement, NodeAnnouncement)>, i64);
```

Comment:
Wait, shouldn't there also be two channel_updates? One for each direction?

Reply:
Yes there is but the function iterates over ChannelInfo, that only keeps the direction stuff. So we will send the out another channel announcement with that update.

Reply:
unless I have channel info wrong?

Reply:
Yea, I think the ChannelUpdate message should be stored per-DirectionalChannelInfo, not per-ChannelInfo

Reply:
Yip definitely did this wrong. I think its fixed now. I only stored it the latest update between the two channels and not each latest one